src/lib.rs

```diff
@@ -251,6 +252,69 @@ where
     )
 }
 
+/// Whether to continue the extraction or not.
+///
+/// Continue(true) will keep going where Continue(false) will stop without uncompressing any
+/// further files.
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub struct Continue(pub bool);
```

Comment:
```rust
enum Callback {
    Continue,
    Stop,
}
```
Might be more meaninful and avoid the use of a boolean inside it.

Reply:
If we keep the callback interface, which I am not against, for now, we'd need to add the support for the async ones as well.

Reply:
I used a newtype as I felt it was more clear and ergonomic but it's more of a style issue. I'll change it if you feel you want to merge this.

I think the advantages from doing this asynchronously are very minor, but it should be simple to as long as the callbacks themselves are Send and not themselves async. Supporting async callbacks would just require the blocking reader thread to block_on every callback result, defeating the purpose and adding unnecessary complexity.